@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -541,6 +546,82 @@ export class PoiRenderer {
         this.m_poiBatchRegistry.update();
     }
 
+    /**
+     * @internal
+     *
+     * Adds a layer to the PoiRenderer
+     * @param layerId
+     */
+    addLayer(layerId: number): PoiLayer {
+        let result = this.getLayer(layerId);
+        if (result === undefined) {
+            result = {
+                id: layerId,
+                scene: new THREE.Scene()
+            };
+
+            this.m_layers.push(result);
+            this.m_layers.sort((a: PoiLayer, b: PoiLayer) => {
+                return a.id - b.id;
+            });
+        }
+        return result;
+    }
+
+    /**
+     * Retrieves a specific `Poi` rendering layer.
+     *
+     * @param layerId - Desired layer identifier.
+     *
+     * @returns Selected {@link PoiLayer}
+     */
+    private getLayer(layerId: number): PoiLayer | undefined {
+        return this.m_layers.find(layer => layer.id === layerId);
+    }
+
+    /**
+     * Renders the content of this `PoiRenderer`.
+     *
+     * @param camera - Orthographic camera.
+     * @param betweenLayersCallback - Optional callback between layers.
+     * @param target - Optional render target.
+     * @param clear - Optional render target clear operation.
+     */
+    render(
+        camera: THREE.OrthographicCamera,
+        betweenLayersCallback?: (
+            lastRendererdLayerId?: number,
+            nextRenderedLayerId?: number
+        ) => void,
+        target?: THREE.WebGLRenderTarget,
+        clear?: boolean
+    ) {
+        let oldTarget: THREE.RenderTarget | null = null;
+        if (target !== undefined) {
+            oldTarget = this.m_renderer.getRenderTarget();
+            this.m_renderer.setRenderTarget(target);
+        }
+        if (clear === true) {
+            this.m_renderer.clear(true);
+        }
+        if (this.m_layers.length <= 0) {
+            betweenLayersCallback?.();
+        } else {
+            for (let i = 0; i < this.m_layers.length; i++) {
+                const layer = this.m_layers[i];
+                if (i === 0) {
+                    betweenLayersCallback?.(undefined, layer.id);
+                }
+                this.update();
+                this.m_renderer.render(layer.scene, camera);
+                betweenLayersCallback?.(layer.id, this.m_layers[i + 1]?.id);
+            }
```

Comment:
The rendering loop probably belongs more to TextElementsRenderer, since it coordinates the rendering of text and icons.
There we could have a straightforward implementation showing the interleaving between pois and text canvases, without the need of abstract callbacks.

Reply:
done